recipes/opencolorio/all/patches/1.1.1.patch

```diff
@@ -31,28 +31,92 @@ index e4f3119..1f7f77c 100644
-     endif()
-     set(TINYXML_CMAKE_ARGS ${TINYXML_CMAKE_ARGS} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
+@@ -168,204 +170,20 @@ messageonce("Setting EXTDIST_PYTHONPATH: ${EXTDIST_PYTHONPATH}")
+ ### tinyxml ###
+ ##TODO: yaml and tinyxml : when there are not USE_EXTERNAL_TINYXML, use the same cmake schema/instructions => maybe refactorize in a cmake functions/macros ?
+ 
+-if(USE_EXTERNAL_TINYXML)
+-    set(TINYXML_VERSION_MIN "2.6.1")
+-    find_package(TinyXML)
+-    if(TINYXML_FOUND)
+-        if(TINYXML_VERSION VERSION_EQUAL ${TINYXML_VERSION_MIN} OR
+-           TINYXML_VERSION VERSION_GREATER ${TINYXML_VERSION_MIN})
+-            message(STATUS "External TinyXML will be used.")
+-        else()
+-            message(FATAL_ERROR "ERROR: ${TINYXML_VERSION} found, but ${TINYXML_VERSION_MIN} or newer is required.")
+-        endif()
+-    else(TINYXML_FOUND)
+-        message(STATUS "TinyXML was not found. Perhaps you forgot to install the development package?")
+-    endif(TINYXML_FOUND)
+-
+-else(USE_EXTERNAL_TINYXML)
+-    set(TINYXML_VERSION 2_6_1)
+-    set(TINYXML_CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${PROJECT_BINARY_DIR}/ext/dist -DOCIO_INLINES_HIDDEN:BOOL=${OCIO_INLINES_HIDDEN})
+-    if(CMAKE_TOOLCHAIN_FILE)
+-        set(TINYXML_CMAKE_ARGS ${TINYXML_CMAKE_ARGS} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE})
+-    endif()
+-    set(TINYXML_CMAKE_ARGS ${TINYXML_CMAKE_ARGS} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
 -    set(TINYXML_ZIPFILE     "${CMAKE_SOURCE_DIR}/ext/tinyxml_${TINYXML_VERSION}.tar.gz")
 -    set(TINYXML_PATCHFILE   "${CMAKE_SOURCE_DIR}/ext/tinyxml_${TINYXML_VERSION}.patch")
 -    set(TINYXML_SOURCE_DIR  "${CMAKE_BINARY_DIR}/ext")
-+    set(TINYXML_ZIPFILE     "${PROJECT_SOURCE_DIR}/ext/tinyxml_${TINYXML_VERSION}.tar.gz")
-+    set(TINYXML_PATCHFILE   "${PROJECT_SOURCE_DIR}/ext/tinyxml_${TINYXML_VERSION}.patch")
-+    set(TINYXML_SOURCE_DIR  "${PROJECT_BINARY_DIR}/ext")
-     ## Create our TINYXML_LIB target
-     if(CMAKE_VERSION VERSION_GREATER "2.8.7")
-         option(TINYXML_OBJECT_LIB_EMBEDDED ${OCIO_BUILD_STATIC} "directly embedded tinyxml 3rdParty object lib into our resulting static lib, no link needed anymore")
-@@ -204,6 +206,7 @@ else(USE_EXTERNAL_TINYXML)
-         set(tinyxml_sources ${TINYXML_SOURCE_DIR}/tinyxml/tinystr.cpp       ${TINYXML_SOURCE_DIR}/tinyxml/tinyxml.cpp 
-                             ${TINYXML_SOURCE_DIR}/tinyxml/tinyxmlerror.cpp  ${TINYXML_SOURCE_DIR}/tinyxml/tinyxmlparser.cpp )
-         set(tinyxml_headers ${TINYXML_SOURCE_DIR}/tinyxml/tinystr.h         ${TINYXML_SOURCE_DIR}/tinyxml/tinyxml.h )
-+        file(MAKE_DIRECTORY "${TINYXML_SOURCE_DIR}")
-         add_custom_command(                                 ## will be done at build time
-             OUTPUT ${tinyxml_sources} ${tinyxml_headers}    ## expected output files
-             COMMAND ${CMAKE_COMMAND} -E tar xzf ${TINYXML_ZIPFILE}
-@@ -248,124 +251,12 @@ endif(USE_EXTERNAL_TINYXML)
+-    ## Create our TINYXML_LIB target
```

Comment:
For smaller patches that are also easier portable to newer versions,
what I often do is wrap deleted code with `if(0)` and `endif()`.
The smaller patches are much more readable.

Reply:
Agreed. I now removed large part of the original patch using `if(0)` and `endif()` as well. Couldn't keep the number of lines the same everywhere, but did my best.